iterators.tex

```diff
@@ -6327,19 +6327,26 @@
 
 \begin{itemdecl}
 template <class T>
-using IteratorType = decltype(begin(declval<T>()));
+  @\newtxt{requires requires (T\& t) \{}@
+    @\newtxt{\{ begin(t) \} -> Iterator; // not required to be equality preserving}@
+  @\newtxt{\}}@
+using IteratorType = decltype(begin(declval<T@\newtxt{\&}@>()));
 
 template <class T>
-using SentinelType = decltype(end(declval<T>()));
+  @\newtxt{requires requires (T\& t) \{}@
+    @\newtxt{typename IteratorType<T>;}@
+    @\newtxt{\{ end(t) \} -> Sentinel<IteratorType<T>{}>; // not required to be equality preserving}@
+  @\newtxt{\}}@
+using SentinelType = decltype(end(declval<T@\newtxt{\&}@>()));
 
 template <class T>
 concept bool Range@\newtxt{() \{}\oldtxt{ =}@
-  @\newtxt{return}@ requires(T@\newtxt{\&\&}@ t) {
+  @\newtxt{return}@ requires@\oldtxt{(T t)}@ {
     typename IteratorType<T>;
     typename SentinelType<T>;
-    { begin(t) } -> @\newtxt{Same<}@IteratorType<T>@\newtxt{>}@; @\newtxt{// not \newtxt{required to be} equality preserving}@
-    { end(t) } -> @\newtxt{Same<}@SentinelType<T>@\newtxt{>}@; @\newtxt{// not \newtxt{required to be} equality preserving}@
-    requires Sentinel<SentinelType<T>, IteratorType<T>>@\newtxt{()}@;
+    @\oldtxt{\{ begin(t) \} -> IteratorType<T>;}@
+    @\oldtxt{\{ end(t) \} -> SentinelType<T>;}@
+    @\oldtxt{requires Sentinel<SentinelType<T>, IteratorType<T>{}>;}@
```

Comment:
Yes - in fact it's implemented that way in cmcstl2. I included both in `Range` to provide an immediate visual link between the three things in the text. It's simply for readability.